### PR TITLE
Improved test coverage for corehq.util.workbook_json.excel

### PR DESCRIPTION
**Pull request by orangejenny**
Follow up for https://github.com/dimagi/commcare-hq/pull/24174

Resurrects the test removed in https://github.com/dimagi/commcare-hq/pull/24174/commits/fbc26ac994a988731a0edd4ce979136a6a2a7035 and adds test coverage for new functions.

### CODE REVIEW

File: corehq/apps/fixtures/upload/workbook.py [hunk 1]
```diff
@@ -26,7 +26,7 @@ class _FixtureWorkbook(object):
     def __init__(self, file_or_filename):
         try:
             self.workbook = excel_get_workbook(file_or_filename)
-        except WorkbookJSONError:
+        except WorkbookJSONError as e:
             raise FixtureUploadError([six.text_type(e)])
```
Comment (emord) [inline corehq/apps/fixtures/upload/workbook.py]: I think you want to change this to `FAILURE_MESSAGES['not_excel_file']` or change the failure message that's expected in the test. There was a different error message when I was testing on my dev branch earlier

Comment (orangejenny) [reply corehq/apps/fixtures/upload/workbook.py]: c63c22ccc58

File: corehq/apps/fixtures/tests/test_upload.py [hunk 2]
```diff
@@ -197,6 +197,10 @@
         'things': [('UID', 'Delete(Y/N)', 'field: name'), (None, 'N', 'apple')],
         'types': [('Delete(Y/N)', 'table_id', 'is_global?', 'field 1'), ('N', 'things', 'yes', 100)]
     }),
+    ('not_excel_file', [
+        "Upload failed! Please make sure you are using a valid Excel 2007 or later (.xlsx) file. " \
+        "Error details: \"There is no item named '[Content_Types].xml' in the archive\".",
```
Comment (emord) [inline corehq/apps/fixtures/tests/test_upload.py]: I'm guessing that this isn't too useful for most people, but its probably being shown in other uploads

**Review by emord — APPROVED** [review]
I fixed a merge conflict with my just merged PR

File: corehq/apps/fixtures/tests/test_upload.py (hunk 2 above)
Comment (orangejenny) [reply corehq/apps/fixtures/tests/test_upload.py]: Agreed the details aren't actionable in this case. I imagine they vary a lot depending on what type of not-Excel content the user uploads.

**Pull request merged**